src/h5web/vis-packs/core/utils.ts

```diff
@@ -180,23 +166,30 @@ export function extendDomain(
     return domain;
   }
 
-  const [min, max] = domain;
-  const isLog = scaleType === ScaleType.Log;
+  const { validMin } = H5WEB_SCALES[scaleType];
 
-  if (min <= 0 && isLog) {
-    throw new Error('Expected domain compatible with log scale');
+  const [min] = domain;
+  if (min < validMin) {
+    throw new Error(`Expected domain compatible with ${scaleType} scale`);
   }
 
+  const extendedDomain = unsafeExtendDomain(domain, extendFactor, scaleType);
+
+  return [Math.max(validMin, extendedDomain[0]), extendedDomain[1]];
+}
+
+function unsafeExtendDomain(
+  domain: Domain,
+  extendFactor: number,
+  scaleType = ScaleType.Linear
```

Comment:
You don't need the default parameter here, since the function is local and used in `extendDomain` only.